src/Native/System.Security.Cryptography.Native/pal_ecdsa_import_export.cpp

```diff
@@ -0,0 +1,461 @@
+	BIGNUM* yBn = nullptr;
+	BIGNUM* pBn = nullptr;
+	BIGNUM* aBn = nullptr;
+	BIGNUM* bBn = nullptr;
+	BIGNUM* nBn = nullptr;
+	BIGNUM* hBn = nullptr;
+	BIGNUM* sBn = nullptr;
+
+	// exit if CryptoNative_GetECKeyParameters failed
+	if (rc != 1) goto error;
+
+	xBn = BN_new();
+	yBn = BN_new();
+	pBn = BN_new();
+	aBn = BN_new();
+	bBn = BN_new();
+	nBn = BN_new();
+	hBn = BN_new();
+
+	if (!xBn || !yBn || !pBn || !aBn || !bBn || !nBn) goto error;
+
+	group = const_cast<EC_GROUP*>(EC_KEY_get0_group(publicKey)); // curve
+	if (!group) goto error;
+
+	curveMethod = const_cast<EC_METHOD*>(EC_GROUP_method_of(group));
+	if (curveMethod == nullptr) 
+        goto error;
+
+	*curveType = MethodToCurveType(curveMethod);
+	if (*curveType == ECCurveType::Unspecified) 
+        goto error;
+
+	// extract P, A, B
+	if (*curveType == ECCurveType::Characteristic2)
+	{
+		if (!EC_GROUP_get_curve_GF2m(group, pBn, aBn, bBn, nullptr)) 
+            goto error;
+	}
+	else
+	{
+		if (!EC_GROUP_get_curve_GFp(group, pBn, aBn, bBn, nullptr)) 
+            goto error;
+	}
+
+	// extract Gx and Gy
+	G = const_cast<EC_POINT*>(EC_GROUP_get0_generator(group));
+	if (*curveType == ECCurveType::Characteristic2)
+	{
+		if (!EC_POINT_get_affine_coordinates_GF2m(group, G, xBn, yBn, NULL)) 
+            goto error;
+	}
+	else
+	{
+		if (!EC_POINT_get_affine_coordinates_GFp(group, G, xBn, yBn, NULL)) 
+            goto error;
+	}
+
+	// extract n
+	if (!EC_GROUP_get_order(group, nBn, nullptr)) 
+        goto error;
+
+	// extract h
+	if (!EC_GROUP_get_cofactor(group, hBn, nullptr)) 
+        goto error;
+
+	// extract S (optional)
+	if (EC_GROUP_get0_seed(group))
+	{
+		sBn = BN_bin2bn(EC_GROUP_get0_seed(group), 
+			static_cast<int>(EC_GROUP_get_seed_len(group)), NULL);
+		// todo: determine how to also extract the nid to the algorithm (eg NID_sha256) that was used with seed
+        *S = sBn;
+        *cbS = BN_num_bytes(sBn);
+	}
+
+	// success; assign variables
+	*Gx = xBn; *cbGx = BN_num_bytes(xBn);
+	*Gy = yBn; *cbGy = BN_num_bytes(yBn);
+	*P = pBn;  *cbP =  BN_num_bytes(pBn);
+	*A = aBn;  *cbA =  BN_num_bytes(aBn);
+	*B = bBn;  *cbB =  BN_num_bytes(bBn);
+	*n = nBn;  *cbn =  BN_num_bytes(nBn);
+	*h = hBn;  *cbh =  BN_num_bytes(hBn);
+
+	rc = 1;
+	goto exit;
+
+error:
+	// clear out variables from CryptoNative_GetECKeyParameters
+	*cbQx = *cbQy = *cbd = 0;
+	*qx = *qy = *d = nullptr;
+
+	// clear our out variables
+	*curveType = ECCurveType::Unspecified;
+	*cbP = *cbA = *cbB = *cbGx = *cbGy = *cbn = *cbh = *cbS = 0;
+	*P = *A = *B = *Gx = *Gy = *n = *h = *S = nullptr;
+
+	if (xBn) BN_free(xBn);
+	if (yBn) BN_free(yBn);
+	if (pBn) BN_free(pBn);
+	if (aBn) BN_free(aBn);
+	if (bBn) BN_free(bBn);
+	if (nBn) BN_free(nBn);
+	if (hBn) BN_free(hBn);
+	if (sBn) BN_free(sBn);
+
+exit:
+	return rc;
+}
+
+extern "C" EC_KEY* CryptoNative_EcKeyCreateByPublicKeyParameters(int32_t nid, uint8_t* qx, int32_t qxLength, uint8_t* qy, int32_t qyLength, uint8_t* d, int32_t dLength)
```

Comment:
The method name says "PublicKey", but it accepts d...  is this just EcKeyCreateByNid?